src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts

```diff
@@ -523,23 +523,20 @@ export class VolumesListTableConfig implements InputTableConf {
           label: T("Delete Dataset"),
           onClick: (row1) => {
 
-            this.dialogService.confirm(T("Delete"), T("This action is irreversible and will \
-             delete any existing snapshots of this dataset (" + row1.path + ").  Please confirm."), false, T('Delete Dataset')).subscribe((res) => {
-                if (res) {
+            this.dialogService.confirm(T("Delete"), 
+              T("This action is irreversible and will delete any existing snapshots of this dataset ") + "(" + row1.path + ")."+  T(" Please confirm.")
+              , false, T('Delete Dataset')).subscribe((confirmed) => {
+                if (confirmed) {
 
                   this.loader.open();
-
-                  const url = "storage/dataset/" + row1.path;
-
-
-                  this.rest.delete(url, {}).subscribe((res) => {
+                  this.ws.call('pool.dataset.delete', [row1.path, {"recursive": true}]).subscribe((wsResp) => {
                     this.loader.close();
                     this.parentVolumesListComponent.repaintMe();
-                  }, (error) => {
+    
+                  }, (e_res) => {
                     this.loader.close();
-                    this.dialogService.errorReport(T("Error deleting dataset."), error.message, error.stack);
+                    this.dialogService.errorReport(T("Error Deleting dataset.") + row1.path, e_res.reason, e_res.stack);
```

Comment:
Don't capitalize random words, and make the dataset name stand out.
```suggestion
                    this.dialogService.errorReport(T("Error deleting dataset ") + "<i>" + row1.path + "</i>.", e_res.reason, e_res.stack);
```